src/Controller/ReadingController.php

```diff
@@ -100,6 +100,10 @@ public function term_form(
         $text = implode($zws, $cleanedparts);
 
         $termdto = $facade->loadDTO($lid, $text);
+        if ($termdto->language->getLgParserType() == 'japanese') {
+            $termdto->Romanization = $termdto->language->
+            getParser()->getReading($text, $termdto->language);
+        }
```

Comment:
A few things here:

- the code belongs in the loadDTO method.  loadDTO should return a complete DTO.
- this replaces the existing Romanization if there is any :-P
- I don't like the "if language ==" branch check, because these kinds of checks can rapidly proliferate in the code.  A more centralized version might do something like the below:

```
// in loadDto:

if (($dto->Romanization ?? '') == '') {
  $dto->Romanization = $lang->getParser()->getRomanization($text);
}

// the different parsers would implement getRomanization().
```